kani_metadata/src/lib.rs

```diff
@@ -10,7 +10,21 @@ pub use vtable::*;
 use serde::{Deserialize, Serialize};
 
 /// The structure of `.kani-metadata.json` files, which are emitted for each crate
-#[derive(Serialize, Deserialize)]
+#[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct KaniMetadata {
+    /// The proof harnesses (#[kani::proof]) found in this crate.
     pub proof_harnesses: Vec<HarnessMetadata>,
+    /// The features found in this crate that Kani does not support.
+    /// (These general translate to `assert(false)` so we can still attempt verification.)
+    pub unsupported_features: Vec<UnsupportedFeature>,
+    /// If crates are built in test-mode, then test harnesses will be recorded here.
+    pub test_harnesses: Vec<HarnessMetadata>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct UnsupportedFeature {
+    /// A string identifying the feature.
+    pub feature: String,
```

Comment:
Can this be an enum?

Reply:
In principle, maybe. But it's a string throughout `kani-compiler`, so that'd be a more extensive refactoring.